tests/Unit/PointwiseFunctions/AnalyticSolutions/NewtonianEuler/Test_RiemannSolver.cpp

```diff
@@ -0,0 +1,204 @@
+                   std::numeric_limits<double>::signaling_NaN());
+  test_solution<3>({{0.0, 0.4, -0.12}}, "[0.0, 0.4, -0.12]",
+                   {{0.0, -0.3, 0.53}}, "[0.0, -0.3, 0.53]", DataVector(5));
+}
+
+// Test correct evaluation of the variables in the star region for the five
+// initial setups listed in Table 4.1 of \Toro2009. The numerical values to
+// compare with are listed in Table 4.2 of the same reference.
+SPECTRE_TEST_CASE(
+    "Unit.PointwiseFunctions.AnalyticSolutions.NewtEuler.RiemannSolver.StarRg",
+    "[Unit][PointwiseFunctions]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{
+      "PointwiseFunctions/AnalyticSolutions/NewtonianEuler"};
+  // The accuracy of the numbers provided by Toro isn't too high so
+  // we need to adjust the precision of the comparison accordingly.
+  Approx larger_approx = Approx::custom().epsilon(1.e-5);
+
+  const double adiabatic_index = 1.4;
+  // Initial position doesn't affect the numbers tested below.
+  const double initial_position = 0.7;
+
+  // Sod or Shock-Tube problem: one shock wave and one rarefaction wave.
+  NewtonianEuler::Solutions::RiemannSolver<1> sod(
+      adiabatic_index, initial_position, 1.0, {{0.0}}, 1.0, 0.125, {{0.0}},
+      0.1);
+  CHECK_ITERABLE_CUSTOM_APPROX(sod.values_at_star_region(),
+                               make_array(0.30313, 0.92745), larger_approx);
+
+  // "123" problem: two strong rarefaction waves. Contact is stationary.
+  NewtonianEuler::Solutions::RiemannSolver<2> one_two_three(
+      adiabatic_index, initial_position, 1.0, {{-2.0}}, 0.4, 1.0, {{2.0}}, 0.4);
+  CHECK_ITERABLE_CUSTOM_APPROX(one_two_three.values_at_star_region(),
+                               make_array(0.00189, 0.),
+                               Approx::custom().epsilon(1.e-2));
+
+  // Left half of the blast wave problem of Woodward&Corella.
+  NewtonianEuler::Solutions::RiemannSolver<3> left_blast_wave(
+      adiabatic_index, initial_position, 1.0, {{0.0}}, 1000.0, 1.0, {{0.0}},
+      0.01);
+  CHECK_ITERABLE_CUSTOM_APPROX(left_blast_wave.values_at_star_region(),
+                               make_array(460.894, 19.5975), larger_approx);
+
+  // Right half of the blast wave problem of Woodward&Corella.
+  NewtonianEuler::Solutions::RiemannSolver<3> right_blast_wave(
+      adiabatic_index, initial_position, 1.0, {{0.0}}, 0.01, 1.0, {{0.0}},
+      100.0);
+  CHECK_ITERABLE_CUSTOM_APPROX(right_blast_wave.values_at_star_region(),
+                               make_array(46.0950, -6.19633), larger_approx);
+
+  // Collision of the two previous shocks.
+  NewtonianEuler::Solutions::RiemannSolver<3> shock_collision(
+      adiabatic_index, initial_position, 5.99924, {{19.5975}}, 460.894, 5.99242,
+      {{-6.19633}}, 46.0950);
+  CHECK_ITERABLE_CUSTOM_APPROX(shock_collision.values_at_star_region(),
+                               make_array(1691.64, 8.68975), larger_approx);
+}
+
+[[noreturn]] SPECTRE_TEST_CASE(
```

Comment:
If you're going to have the assertion tests can you please add an output regex that is check to make sure you triggered the correct assertion?